src/transport/http2.rs

```diff
@@ -192,23 +183,11 @@ impl<In: RpcMessage, Out: RpcMessage> ServerChannel<In, Out> {
 
     /// Creates a server listening on the [`SocketAddr`] with a custom configuration.
     pub fn serve_with_config(addr: &SocketAddr, config: ChannelConfig) -> hyper::Result<Self> {
-        let mut addr_incoming = AddrIncoming::bind(addr)?;
-        addr_incoming.set_nodelay(true);
-        Self::serve_with_incoming(addr_incoming, Arc::new(config))
-    }
-
-    /// Serve with a custom incoming and custom config
-    fn serve_with_incoming<I>(incoming: I, config: Arc<ChannelConfig>) -> hyper::Result<Self>
-    where
-        I: hyper::server::accept::Accept + Send + 'static,
-        I::Conn: AsyncRead + AsyncWrite + Unpin + Send + HasRemoteAddr + 'static,
-        I::Error: Into<Box<dyn error::Error + Send + Sync>>,
-    {
         let (accept_tx, accept_rx) = flume::bounded(32);
 
         // The hyper "MakeService" which is called for each connection that is made to the
         // server.  It creates another Service which handles a single request.
-        let service = make_service_fn(move |socket: &I::Conn| {
+        let service = make_service_fn(move |socket: &AddrStream| {
```

Comment:
This explicitly reverts this generic stuff.  We need this transport to be HTTP2 and for that we need this to be TCP and that means we have remote and local addresses.

Reply:
So how do you propose doing UDS? Not at all?

Reply:
heh, I'm surprised you merged this :)

Originally I was thinking it should be a separate module, but over the holidays I started thinking it perhaps is not the best idea.  I'll try and make this generic again.

Reply:
I was not entirely happy with having to add the generics stuff. Maybe it will benefit from us both sharing the pain...